pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -367,7 +370,8 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 	for i := range replicas {
 		// delete and recreate failed pods
 		if isFailed(replicas[i]) {
-			glog.V(4).Infof("StatefulSet %s/%s is recreating failed Pod %s",
+			ssc.recorder.Eventf(set, v1.EventTypeWarning, "RecreatingFailedPod",
```

Comment:
if the pod fail after recreate, because no nodes are available or something is permanently broken in the pod (like a service it depends on is down), how often will this event occur?

Reply:
It should be a one off. Pods with restart policy Always can fail on node down, but won't get scheduled there again as the node is down.